### ISSUE TITLE
Bug: Person shows "No options" in standalone

### ISSUE THREAD

**Issue by chmac** [mattermost/focalboard#1026]
**Summary:**
In focalboard standalone, the property type `Person` shows only `No options`.

**Steps to reproduce the behavior:**
1. Install focalboard standalone
2. Create a card, add a property of type Person
3. See `No options`

**Expected behavior:**

From #559 it seems like this list is populated from the mattermost users. I'd assumed that there was somewhere a list of `Person`s that I could edit. But maybe the solution is to simply disable this property type in the standalone version?

**Comment by jespino** [comment]
This is working fin on the main branch, can you test it against the last 0.9.0-rc5?

**Comment by chmac** [comment]
@jespino When you say "working", can you elaborate on what that means? Is there a place where I can edit the list of `Person`s in the standalone app?

How would I test the 0.9.0-rc5 version? I'm running the standalone (macOS App Store installed) version of focalboard. This issue only applies to the **standalone** version of the app as far as I'm aware.

**Comment by jespino** [comment]
Ohh, you are talking about the App, that is different, the app is "single" user, so there is no `Person` to list, you can have different "Person" in a personal server, that is bind to users.

For Desktop App, the right approach is to have a select field, and manually introduce the different persons. The `Person` type field is for environments with multiple users. Maybe we should hide it in the Desktop App.

**Comment by jespino** [comment]
@chenilim what do you think about it?

**Comment by chmac** [comment]
@jespino I thought that might be the solution. One question that comes up is what happens with data export / import, if the data has a value of "Person" then what happens to it? What is the underlying data stored in that property? Is it an ID of some sort?

What about changing the implementation of the Person property so that on the standalone desktop app it behaves like a Select?

Then if the data is exported / imported, it would show as the ID instead of the data being simply lost...

**Comment by jespino** [comment]
If that value is Person, it stores the `userId`, what is the udnerlaying data stored in the property.

The option of changing it to a select on standalone version sounds right, and yes, the "matching" is not going to be great, because the user ID is not a very good identifier, anyway we can change how we export/import that in the future, because makes more sense to use something more widely unique but widely stable at the same time, something like the email address.

@chenilim what do you think about this proposed fix?

**Comment by chenilim** [comment]
Yup, I agree in principle. Though perhaps it would be easier to convert to a text property instead (which can be explicitly converted to a Select property by the user). The value would ideally be populated with the userID (i.e. username or email). This conversion should happen at import time.

Agree that the person property type should be disabled when running in standalone (App) mode. i.e. remove the Person property type as an option in the menu.

**Comment by chenilim** [comment]
FYI @asaadmahmood, any additional thoughts? More generally, we should rethink the role of export / import archive. The original intent was as a convenience to developers and a quick way to migrate among editions. However, export / import is more of a Sys Admin feature in larger multi-user environments. It's a powerful feature that can lead to problems like data loss if used incorrectly, or with the wrong expectations.

**Comment by chmac** [comment]
> Yup, I agree in principle. Though perhaps it would be easier to convert to a text property instead (which can be explicitly converted to a Select property by the user). The value would ideally be populated with the userID (i.e. username or email). This conversion should happen at import time.

If you're importing into the desktop app you won't have the user table so you won't be able to convert `userId` into email or username.

Personally I'd suggest that converting it to a String field and leaving it as the `userId` is a totally acceptable solution without too much effort. There's no elegant solution, if usernames or emails change then reimporting the same date wouldn't work any more. While the `userId` should be consistent.

I'd suggest that just rendering the Person type as a string in the app would also allow the data to be exported and imported without data loss. While if it's really converted to a string, that's a destructive operation.
